core/src/main/java/co/aikar/commands/CommandManager.java

```diff
@@ -377,6 +380,71 @@ public void addSupportedLanguage(Locale locale) {
         getLocales().loadMissingBundles();
     }
 
+    /**
+     * Registers an instance of a class to be registered as an injectable dependency.<br>
+     * The command manager will attempt to inject all fields in a command class that are annotated with
+     * {@link co.aikar.commands.annotation.Dependency} with the provided instance.
+     *
+     * @param clazz the class the injector should look for when injecting
+     * @param instance the instance of the class that should be injected
+     * @throws IllegalArgumentException when instance isn't an instance of clazz
+     * @throws IllegalStateException when there is already an instance for the provided class registered
+     */
+    public void registerDependency(Class<?> clazz, Object instance){
```

Comment:
Use <T> with Class <T> clazz, T instance, to require it actually satisfies the class its providing for.

Reply:
generic T, github stripped it...

Reply:
```java
public <T> void registerDependency(Class<T> clazz, T instance) {
```

Reply:
I had this and it didn't work. when I want to pass the plugins class, I can't cast to it, thus throwing a compiler error. https://i.imgur.com/ynHskze.png